src/communication/network_interface_sr.c

```diff
@@ -7268,6 +7269,385 @@ slogtb_dump_trantable (THREAD_ENTRY * thread_p, unsigned int rid, char *request,
+
+  mmon_aggregate_server_info (server_info);
+
+  size += or_packed_string_length (server_info.name, NULL);
+  // Size of total_mem_usage
+  size += OR_INT64_SIZE;
+
+  buffer = (char *) db_private_alloc (thread_p, size);
+
+  if (buffer == NULL)
+    {
+      er_set (ER_ERROR_SEVERITY, ARG_FILE_LINE, ER_OUT_OF_VIRTUAL_MEMORY, 1, size);
+      error = ER_OUT_OF_VIRTUAL_MEMORY;
+    }
+  else
+    {
+      ptr = or_pack_string (buffer, server_info.name);
+      ptr = or_pack_int64 (ptr, server_info.total_mem_usage);
+    }
+
+  if (error != NO_ERROR)
+    {
+      ptr = or_pack_int (reply, 0);
+      ptr = or_pack_int (ptr, error);
+      css_send_data_to_client (thread_p->conn_entry, rid, reply, OR_ALIGNED_BUF_SIZE (a_reply));
+    }
+  else
+    {
+      ptr = or_pack_int (reply, size);
+      ptr = or_pack_int (ptr, error);
+      css_send_data_to_client (thread_p->conn_entry, rid, reply, OR_ALIGNED_BUF_SIZE (a_reply));
+      css_send_data_to_client (thread_p->conn_entry, rid, buffer, size);
+    }
+  db_private_free_and_init (thread_p, buffer);
+}
+
+/*
+ * smmon_get_module_info -
+ *
+ * return:
+ *
+ *   rid(in):
+ *   request(in):
+ *   reqlen(in):
+ */
+void
+smmon_get_module_info (THREAD_ENTRY * thread_p, unsigned int rid, char *request, int reqlen)
+{
+  char *buffer, *ptr;
+  int size = 0;
+  OR_ALIGNED_BUF (OR_INT_SIZE + OR_INT_SIZE) a_reply;
+  char *reply = OR_ALIGNED_BUF_START (a_reply);
+  int error = NO_ERROR;
+  MMON_MODULE_INFO *module_info;
+  MMON_COMP_INFO *comp_info;
+  MMON_SUBCOMP_INFO *subcomp_info;
+  int module_index, module_count;
+
+  ptr = or_unpack_int (request, &module_index);
+
+  if (module_index == 0)
+    {
+      module_info = (MMON_MODULE_INFO *) malloc (sizeof (MMON_MODULE_INFO) * MMON_MODULE_LAST);
+      module_count = MMON_MODULE_LAST;
+    }
+  else
+    {
+      module_info = (MMON_MODULE_INFO *) malloc (sizeof (MMON_MODULE_INFO));
+      module_count = 1;
+    }
+
+  if (module_info == NULL)
+    {
+      er_set (ER_ERROR_SEVERITY, ARG_FILE_LINE, ER_OUT_OF_VIRTUAL_MEMORY, 1,
+	      sizeof (MMON_MODULE_INFO) * MMON_MODULE_LAST);
+      error = ER_OUT_OF_VIRTUAL_MEMORY;
+    }
+  else
+    {
+      error = mmon_aggregate_module_info (module_info, module_index);
+
+      if (error == NO_ERROR)
+	{
+	  /* before send information 
+	   * 1) check size
+	   * 2) allocate buffer
+	   * 3) packing information 
+	   * 4) deallocate MMON_MODULE_INFO */
+
+	  /* 1) check size 
+	   *    - calculate buffer size of packed values
+	   *    - module info -> component info w/ subcomponent info */
+	  for (int idx = 0; idx < module_count; idx++)
+	    {
+	      size += or_packed_string_length (module_info[idx].name, NULL);
+	      // Size of stat information (OR_INT64_SIZE * 3 + OR_INT_SIZE)
+	      // and size of num_comp (OR_INT_SIZE)
+	      size += OR_INT64_SIZE * 3 + OR_INT_SIZE * 2;
+
+	      for (uint32_t i = 0; i < module_info[idx].num_comp; i++)
+		{
+		  comp_info = &(module_info[idx].comp_info[i]);
```

Comment:
```suggestion
	      for (const &auto comp_info : module_info[idx].comp_info;)
		{
```

Reply:
The same goes for all subcom_info and com_info here in this function.

Reply:
Done.